### DISCUSSION TITLE
Suggestion: apply indentation only to a certain depth

### DISCUSSION THREAD

**Discussion by david-topham-precisely** [nlohmann/json#3471] [Ideas]
Consider the following JSON:

`{"name": "thing1", "vals": [34,45,66], "obj" : {"a":"ice","b":"snow"}}`

If I use the basic_json::dump() function with the indentation set to a positive value, e.g. dump(4). I get indented output:

```
{
   "name": "thing1",
   "vals": [
      34,
      45,
      66
   ],
   "obj": {
      "a": "ice",
      "b": "snow"
   }
}
```

This is fine as it is: using the indentation makes it readable. But it also makes it take up 12 lines, and I may have 1000s of these in an array that I want to write to a file. 

I may know that the **vals** array will always contain a small number of values; I may also know that the **obj** sub-object will only contain two short items. It would be good to be able to specify to what depth in the indentation should apply, which would result in a more compact form:

```
{
   "name": "thing1",
   "vals": [34, 45, 66],
   "obj": {"a": "ice", "b": "snow"}
}
```
In this case, the indentation / line breaks are applied to the top-level object, but not to the sub-items. 

I can't see any way to do that at present (please correct me if I'm wrong).

The idea / suggestion is to consider adding that ability. I'd be interested to know if anyone else considers that would be useful.

**Comment by falbrechtskirchinger** [comment]
A few weeks ago, I went down a rabbit hole and discovered that many people had many different requirements for the serializer over the years. Several solutions were proposed, but none were completed and merged.

It's on my wishlist but not something I have time for. if I did, I'd start by implementing a SAX-like interface for serialization and then try to find a default implementation that would cover most use cases.

**Reply by david-topham-precisely** [reply]
That's cool, just wanted to put it out there. The SAX-like approach sounds like it would be very flexible. 

I can write something that meets my current needs by iterating over the object.
